### ISSUE TITLE
Softmax operator shape inference

### ISSUE THREAD

**Issue by roywei** [awslabs/keras-apache-mxnet#233]
after https://github.com/apache/incubator-mxnet/pull/14661 and https://github.com/apache/incubator-mxnet/pull/14869. 
Softmax operator has more strict partial shape inference. This will result in the ndim() of softmax input maybe unknown during dimension check. It will be inferred at a later time.

I have created PR for the case of `ndim == 0` in for mxnet in https://github.com/awslabs/keras-apache-mxnet/pull/232. 
I'm keeping this issue open to see if we can find alternatives of modifying Keras front end code.
